### PR TITLE
Kills 900+ lines of useless chemistry recipe code

### PR DESCRIPTION
**Pull request by Tsar-Salat**
## About The Pull Request

Removes all name and id defines from chemical reactions

## Why It's Good For The Game

On TG, id and name were briefly used for their randomized chemistry recipe system, and shortly after replaced entirely.

On Bee, I couldnt find a single thing that actually called ID or Name.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Creating fluorosulfuric acid & using it on a machine

https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/4af7977a-caa7-4108-b4c0-8482600efc1d

</details>

## Changelog
:cl: rkz
code: removed 900+ useless chemistry recipe lines
/:cl:

### CODE REVIEW

File: code/modules/reagents/chemistry/holder.dm [hunk 1]
```diff
@@ -30,9 +30,11 @@
 		var/datum/chemical_reaction/D = new path()
 		var/list/reaction_ids = list()
 
-		if(D.required_reagents && D.required_reagents.len)
-			for(var/reaction in D.required_reagents)
-				reaction_ids += reaction
```
Comment (Tsar-Salat) [inline code/modules/reagents/chemistry/holder.dm:35]: Seemingly neither holder.dm or recipes.dm have any references to var/name or var/id???

Why do these vars exist?

**Comment by EvilDragonfiend** [comment]
No. :key:
It is true that nothing usss those, but it is actually correct.
* id: this should be the true id rather than typepath. Typepath being id is a bad practice, but we can't fix it right now.
* name: it might bd fine to remove, but it is better to keep with id since id is not a good format to tell a namd.

I know thess lines are useless and why you want to remove, but reviving these lines when we make an actual fix will be tedious.

File: code/modules/reagents/chemistry/holder.dm (hunk 1 above)
Comment (EvilDragonfiend) [reply code/modules/reagents/chemistry/holder.dm:35]: It is supposed to be used, but old coders did something incorrect.

**Comment by itsmeow** [comment]
I'm not really sure there is any value in removing these.

**Comment by EvilDragonfiend** [comment]
> I'm not really sure there is any value in removing these.

It is inherently dangerous for maintaining in the future. If someone tries to do something complex to the recipe system, they will put the unique dependency on its typepath, but it is not a good practice. Typepath is only a prototype definition, and its uniqueness isn't truly reliable.
Although our current code wouldn't have the issue by removing this, but still, it's not good in the future.

**Comment by itsmeow** [comment]
:key:

Yeah, I'd rather have these around just in case. They don't really do that much harm memory or init wise and they're nice to have if we want to mess around with chem.

**Comment by Tsar-Salat** [comment]
"Just in case" & "when we make a fix" is nonsense for something no one has touched in 5 years.

I'm going to recoup #8579 within the month, after which I will revisit this PR.

**Pull request closed without merging**